apps/rush-lib/src/logic/InstallManager.ts

```diff
@@ -897,6 +905,16 @@ export class InstallManager {
           // Example: "C:\MyRepo\common\temp\npm-local\node_modules\.bin\npm"
           const packageManagerFilename: string = this._rushConfiguration.packageManagerToolFilename;
 
+          const packageManagerEnv: NodeJS.ProcessEnv = process.env;
+
+          if (this._rushConfiguration.packageManagerOptions &&
+            this._rushConfiguration.packageManagerOptions.environmentVariables) {
+
+            for (const envVar of Object.keys(this._rushConfiguration.packageManagerOptions.environmentVariables)) {
+              packageManagerEnv[envVar] = this._rushConfiguration.packageManagerOptions.environmentVariables[envVar];
```

Comment:
What if the process's environment already has an env variable that's specified in the configuration? I'd think the local environment should take precedence. Maybe we should print a warning?

Reply:
I will add a warning. I wanted the rush config to override any setting on the machine by default.

I think you are right about the precedence. How about we support an optional flag `override` in rush config for each environment variable, with default value `false`? That would make it configureable and explicit.

Reply:
I like that.

Reply:
Done